torchvision/prototype/features/_feature.py

```diff
@@ -57,5 +57,24 @@ def new_like(cls, other, data, *, dtype=None, device=None, **metadata):
             metadata.setdefault(name, getattr(other, name))
         return cls(data, dtype=dtype or other.dtype, device=device or other.device, **metadata)
 
+    @classmethod
+    def __torch_function__(
+        cls,
+        func: Callable[..., torch.Tensor],
+        types: Tuple[Type[torch.Tensor], ...],
+        args: Sequence[Any] = (),
+        kwargs: Optional[Mapping[str, Any]] = None,
+    ) -> torch.Tensor:
```

Comment:
Technically, return type should be `Union[torch.Tensor, F]` but this has no practical use. Since the return type depends in the input argument `func`, there is no way for `mypy` to figure out which type it is going to get. So if we would be calling this explicitly, we needed to `cast` anyway if we know we get a `Feature` type back. Good thing for us, we never call this directly. Thus, we can get away with the `Tensor` annotation, since all `Feature`'s are also `Tensor`'s.

Reply:
Thanks for the input. I understand now. Questions:
- Should we remove the F type definition on top then?
- Or perhaps it is still worth it to return F and do an explicit cast, just to reinforce to the reader of the code that this is supposed to return Features? (the argument here is that code is read more times than written)

Reply:
> Should we remove the F type definition on top then?

I've added annotations to the other methods. Now `F` is used.

> Or perhaps it is still worth it to return F and do an explicit cast

This method is never called by us directly, so there no place where we could perform an explicit cast. `__torch_function__` is for example triggered if you do something like `Image() + 1` or any other PyTorch operation. I've added an explanation as docstring. Let me know if you feel it is not sufficient.

Reply:
> I still don't see it being used on this file. Am I missing something?

You are probably only looking at the changes in this PR. I've added the annotations to the base branch, for example https://github.com/pytorch/vision/blob/ab11f617e09733768b30a80a552e8fae2e5de983/torchvision/prototype/features/_feature.py#L40-L46